Feat(eslint-plugin-fiori-tools): add rule for cloud dev adaptation status definition (#4993)
SAP/open-ux-tools: packages/eslint-plugin-fiori-tools/docs/rules/sap-cloud-dev-adaptation-status.md

# sap-cloud-dev-adaptation-status

## Rule Details

> This rule ensures that `cloudDevAdaptationStatus` is defined in the `sap.fiori` section of `manifest.json`.

Reviewer: ```suggestion
This rule ensures that `cloudDevAdaptationStatus` is defined in the `sap.fiori` section of the `manifest.json` file.
```

> The `cloudDevAdaptationStatus` property represents the release status for the developer adaptation in the cloud. The supported types are `released` - app is fully open for cloud adaptation projects and variant generation, `deprecated` - marks older extension pathways that are phased out or no longer recommended, `obsolete` - the app is locked and not enabled for cloud-ready adaptation.

Reviewer: ```suggestion
The `cloudDevAdaptationStatus` property represents the release status for the developer adaptation in the cloud.

The following types are supported: 
- `released`: The app is fully open for cloud adaptation projects and variant generation,.
- `deprecated`: Marks older extension pathways that are phased out or no longer recommended.
- `obsolete`: The app is locked and not enabled for cloud-ready adaptation.
```

## Why Was This Introduced?

> S/4HANA applications are required to declare their cloud dev adaptation status.

Reviewer: ```suggestion
SAP S/4HANA applications are required to declare their cloud dev adaptation status.
```

## Warning Examples

> The following configuration will trigger a warning because `cloudDevAdaptationStatus` is absent:

Reviewer: ```suggestion
The following configurations trigger a warning because `cloudDevAdaptationStatus` is absent:
```

```json
{
  "sap.fiori": {
    "registrationIds": [],
    "archeType": "transactional"
  }
}
```

**Warning message:** "The application hasn't set a release status for the developer adaptation in the cloud."

## Correct Pattern

```json
{
  "sap.fiori": {
    "registrationIds": [],

>     "archeType": "transactional",

Author: It is spelled the same way as in manifest.json.

    "cloudDevAdaptationStatus": "released"
  }
}
```

## How to Fix

> Add a `cloudDevAdaptationStatus` entry inside the `sap.fiori` section of your `manifest.json` with a selected value.

Reviewer: ```suggestion
Add a `cloudDevAdaptationStatus` entry inside the `sap.fiori` section of your `manifest.json` file with a selected value.
```

# sap-cloud-dev-adaptation-status

## Rule Details

This rule ensures that `cloudDevAdaptationStatus` is defined in the `sap.fiori` section of the `manifest.json` file.

The `cloudDevAdaptationStatus` property represents the release status for the developer adaptation in the cloud.
The following types are supported: 

> - `released`: The app is fully open for cloud adaptation projects and variant generation,.

Reviewer 2: remove trailing comma

Author: Thanks, removed.
